dm/syncer/syncer.go

```diff
@@ -2185,7 +2185,7 @@ func (s *Syncer) initSafeModeExitTS(firstBinlogTS int64) error {
 		return err
 	}
 	s.firstMeetBinlogTS = &firstBinlogTS
-	exitTS := firstBinlogTS + int64(duration.Seconds())
+	exitTS := firstBinlogTS + int64(duration.Seconds())*1000
```

Comment:
why need *1000?

Reply:
yes, `firstBinlogTS` which comes from Timestamp is the value of millisecond，but duration is second. i find it when i run integration test.

Reply:
you can use duration.Milliseconds() then

Reply:
https://dev.mysql.com/doc/internals/en/binlog-event-header.html this should be the unit of second, not millisecond 🤔

Reply:
sorry, I have tried again, the value is `uint32`, i will fix back. I may modify this because I encounter another problem，the ts may be 0 when the event type is `ROTATE_EVENT` and then i modify here and avoid 0, but here is wrong.